instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -169,4 +172,10 @@ public OpenTelemetry openTelemetry() {
       return OpenTelemetry.noop();
     }
   }
+
+  @Bean
+  @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+  public static OpenTelemetrySupplier openTelemetrySupplier(BeanFactory beanFactory) {
```

Comment:
Could you please add a comment about the why?

Reply:
comment added